pkg/exemplars/multitsdb.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package exemplars
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/thanos-io/thanos/pkg/exemplars/exemplarspb"
+)
+
+// MultiTSDB implements exemplarspb.ExemplarsServer that allows to fetch exemplars a MultiTSDB instance.
+type MultiTSDB struct {
+	tsdbExemplarsServers func() map[string]exemplarspb.ExemplarsServer
+}
+
+// NewMultiTSDB creates new exemplars.MultiTSDB.
+func NewMultiTSDB(tsdbExemplarsServers func() map[string]exemplarspb.ExemplarsServer) *MultiTSDB {
+	return &MultiTSDB{
+		tsdbExemplarsServers: tsdbExemplarsServers,
+	}
+}
+
+// Exemplars returns all specified exemplars from a MultiTSDB instance.
+func (m *MultiTSDB) Exemplars(r *exemplarspb.ExemplarsRequest, s exemplarspb.Exemplars_ExemplarsServer) error {
+	fmt.Println("meer paas request aayi", m.tsdbExemplarsServers())
+	for tenant, es := range m.tsdbExemplarsServers() {
```

Comment:
Why is this a tenant? Isn't it just an instance that supports tsdb/exemplars?

Reply:
Well yeah, but it's an instance for a particular tenant. Maybe a better name is `tenantExemplarServer` or simply `exemplarServer`?

Reply:
I have a similar question. How would an exemplar query perform for a single tenant? Looks like each exemplar query request would check all tenants' exemplar storage. It is a little bit expensive.

Reply:
Ah, there are some differences here. With a series call, you simply get `[]storepb.LabelMatcher` / `[]*labels.Matcher`, while with exemplars, you get `[][]*labels.Matcher`, where you have a union between the outer slices and then similar to Series call, intersection between the inner `[]*labels.Matcher`. WDYT should be the correct approach here?

I was thinking of looping over the outer array, and filter out `[]*labels.Matcher`s that doesn't match with the external labels.

Reply:
I have pushed a commit with this approach, LMK if this isn't what you meant.